server/routes/api/integrations/integrations.ts

```diff
@@ -127,15 +102,27 @@ router.post(
 router.post(
   "integrations.delete",
   auth({ admin: true }),
-  async (ctx: APIContext) => {
-    const { id } = ctx.request.body;
-    assertUuid(id, "id is required");
-
+  transaction(),
+  validate(T.IntegrationsDeleteSchema),
+  async (ctx: APIContext<T.IntegrationsDeleteReq>) => {
+    const { id } = ctx.input.body;
     const { user } = ctx.state.auth;
-    const integration = await Integration.findByPk(id);
+    const { transaction } = ctx.state;
+
+    const integration = await Integration.findByPk(id, { transaction });
     authorize(user, "delete", integration);
 
-    await integration.destroy();
+    await integration.destroy({ transaction });
+    // also remove the corresponding authentication if it exists
```

Comment:
This is already taken care of through delete cascade on the db relationship

Reply:
That works the other way around, when you delete the `authetications` row. I added a test for this, comment these lines out and there fails the test 🙂